python/flow_sdk/shim_airbyte_cdk.py

```diff
@@ -97,18 +92,21 @@ def to_flow_state(self, index: dict[t.Tuple[str | None, str], t.List[int]], bind
     re-generate the index here as it's possible that binding indices have changed
     """
     @staticmethod
-    def from_flow_state(state: t.Dict[str, t.Any], bindings: t.List[CaptureBinding]):
-        state = State(**{
-            "bindingStateV1": state.get("bindingStateV1", {})
-        })
-
-        index: dict[t.Tuple[str | None, str], t.List[int]] = {}
-        for i, binding in enumerate(bindings):
-            if binding["stateKey"] in state.bindingStateV1:
-                namespace = binding["resourceConfig"].get("namespace",None)
-                stream = binding["resourceConfig"]["stream"]
-                index[(namespace, stream)] = state.bindingStateV1[binding["stateKey"]].rowId
-        return (state, index)
+    def from_flow_state(state: dict | None, bindings: t.List[CaptureBinding]):
+        try:
+            parsed_state = state and State(**state) or State()
+
+            index: dict[t.Tuple[str | None, str], t.List[int]] = {}
+            for i, binding in enumerate(bindings):
+                if binding["stateKey"] in parsed_state.bindingStateV1:
+                    namespace = binding["resourceConfig"].get("namespace",None)
+                    stream = binding["resourceConfig"]["stream"]
+                    index[(namespace, stream)] = [i, parsed_state.bindingStateV1[binding["stateKey"]].rowId]
+            return (parsed_state, index)
+
+        except Exception as e:
```

Comment:
AFAIK this is no longer needed, and for that reason I think we should take it out now.

Reply:
I left the try-catch in so we have a useful error message, but now the exception is re-raised

Reply:
Sounds good, thanks.

FYI there's already some top-level code that catches exceptions and logs them out as structured errors.